sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_options.hpp

```diff
@@ -380,6 +439,29 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     PathAccessConditions AccessConditions;
   };
 
+  using ScheduleFileExpiryOriginType = Blobs::Models::ScheduleBlobExpiryOriginType;
+
+  /**
+   * @brief Optional parameters for #Azure::Storage::Files::DataLake::FileClient::UploadFrom.
+   */
+  struct ScheduleFileDeletionOptions final
+  {
+    /**
+     * The expiry time from the specified origin. Only work if ExpiryOrigin is
+     * ScheduleFileExpiryOriginType::RelativeToCreation or
+     * ScheduleFileExpiryOriginType::RelativeToNow.
+     */
+    Azure::Nullable<std::chrono::milliseconds> TimeToExpire;
+
+    /**
+     * The expiry time in RFC1123 format. Only works if ExpiryOrigin is
+     * ScheduleFileExpiryOriginType::Absolute.
+     */
+    Azure::Nullable<DateTime> ExpiresOn;
+  };
+
+  using SchedulePathDeletionOptions = ScheduleFileDeletionOptions;
```

Comment:
Does ScheduleDeletion also work for directory?

Reply:
No, it's only for file.